lib/posix/options/Kconfig.mem

```diff
@@ -50,7 +50,7 @@ config POSIX_MEMLOCK
 config POSIX_MEMLOCK_RANGE
 	bool "POSIX range memory locking"
 	imply MMU if (CPU_HAS_MMU && ARCH_HAS_DEMAND_PAGING)
-	imply DEMAND_PAGING
+	imply DEMAND_PAGING if !(X86 && !X86_64 && USERSPACE)
```

Comment:
should we create an enhancement issue for these exclusions (when this PR is merged)?

Reply:
Probably some kind of issue